### PR TITLE
[gitpod-shared] Use "waitForSnapshot" to avoid disconnects

### PR DESCRIPTION
**Pull request by geropl**
Companion PR for: https://github.com/gitpod-io/gitpod/pull/6144

The plan is to hold this PR until the other one :point_up: has been deployed.

### CODE REVIEW

File: extensions/gitpod-shared/src/features.ts [hunk 1]
```diff
@@ -474,12 +475,28 @@ export async function registerWorkspaceCommands(context: GitpodExtensionContext)
 			properties: { action: 'snapshot' }
 		});
 		try {
-			const snapshotId = await vscode.window.withProgress({
+			let snapshotId = '';
+			await vscode.window.withProgress({
```
Comment (akosyakov) [inline extensions/gitpod-shared/src/features.ts:479]: it looks like if a user press cancel still, either error or success notification with bogus link will be shown? Should not we guard?

Comment (geropl) [reply extensions/gitpod-shared/src/features.ts:479]: The code should be fine if we assume that `withProgress` bubbles up errors. But it's better to make that explicit, I agree.

**Comment by akosyakov** [comment]
@geropl could you rebase and resolve conflicts please? 🙏 or would you be fine if i do it?

Server side already deployed in production correct?

**Comment by geropl** [comment]
> could you rebase and resolve conflicts please

Uh, that was quick - did rebase yesterday :see_no_evil:

> Server side already deployed in production correct?

Yes, since last week.

**Comment by geropl** [comment]
@akosyakov Should be up-to-date :+1:

**Review by akosyakov — APPROVED** [review]
worked nicely, should be available in tomorrow insiders

**Pull request merged**